src/lib.rs

```diff
@@ -142,21 +142,65 @@ pub struct PrivateKeyRotationCheckResult {
 }
 
 impl PrivateKeyRotationCheckResult {
-    pub fn user_rotation_needed(&self) -> Option<UserId> {
+    pub fn user_rotation_needed(&self) -> Option<&UserId> {
         match &self.rotations_needed {
-            EitherOrBoth::Left(u) | EitherOrBoth::Both(u, _) => Some(u.clone()),
+            EitherOrBoth::Left(u) | EitherOrBoth::Both(u, _) => Some(u),
             _ => None,
         }
     }
 
-    pub fn group_rotation_needed(&self) -> Option<Vec1<GroupId>> {
+    pub fn group_rotation_needed(&self) -> Option<&Vec1<GroupId>> {
         match &self.rotations_needed {
-            EitherOrBoth::Right(groups) | EitherOrBoth::Both(_, groups) => Some(groups.clone()),
+            EitherOrBoth::Right(groups) | EitherOrBoth::Both(_, groups) => Some(groups),
             _ => None,
         }
     }
 
-    // pub fn rotate_all(ironoxide: &IronOxide) -> (UserId, Vec<GroupId>) //TODO consider when group private key rotation is added
+    pub fn rotate_all(
```

Comment:
How does this method end up getting documented? We should have some fairly big warnings about how long this method might take. It's probably unlikely, but if there was a half-dozen groups to rotate and each one had a dozen admins in it, this method will take a looooong time.

Reply:
I added a comment that says that this could take a long time, and strongly recommending using the regular function for individual groups if possible. Let me know if you would change/add to it.

Reply:
Just a thought: If the app/runtime has the ability to execute the rotations in parallel, it probably won't be that bad.